Neos.EventSourcedNeosAdjustments/Classes/NodeImportFromLegacyCR/Command/ContentRepositoryMigrateCommandController.php

```diff
@@ -24,25 +31,56 @@ class ContentRepositoryMigrateCommandController extends CommandController
 {
 
     /**
-     * @Flow\Inject
-     * @var ContentRepositoryExportService
+     * @Flow\InjectConfiguration(path="EventStore.stores.ContentRepository", package="Neos.EventSourcing")
+     * @var array
      */
-    protected $contentRepositoryExportService;
+    protected $eventStoreConfiguration;
 
     /**
-     * @Flow\Inject
-     * @var ImportProjectionPerformanceService
+     * @Flow\Inject(lazy=false)
+     * @var GraphProjector
      */
-    protected $importProjectionPerformanceService;
+    protected $graphProjector;
+
+    /**
+     * @Flow\Inject(lazy=false)
+     * @var WorkspaceProjector
+     */
+    protected $workspaceProjector;
+
+    /**
+     * @Flow\Inject(lazy=false)
+     * @var ContentStreamProjector
+     */
+    protected $contentStreamProjector;
 
     /**
      * Run a CR export
      */
     public function runCommand()
     {
-        $this->contentRepositoryExportService->reset();
-        $this->importProjectionPerformanceService->configureGraphAndWorkspaceProjectionsToRunSynchronously();
-        $this->contentRepositoryExportService->migrate();
+        /** @noinspection PhpMethodParametersCountMismatchInspection */
+        /** @var EventStorageInterface $eventStoreStorage */
+        $eventStoreStorage = $this->objectManager->get($this->eventStoreConfiguration['storage'], $this->eventStoreConfiguration['storageOptions'] ?? []);
+
+        $eventPublisher = new ClosureEventPublisher();
+        $eventStore = new EventStore($eventStoreStorage, $eventPublisher);
```

Comment:
We can't replace the event Publisher for the running event store (for good reasons) so we have to build the instance manually here.. Maybe there's a nicer way to achieve this without so much duplicated logic

Reply:
Please add a comment explaining what we are doing here and WHY ;)